src/main/java/org/openrewrite/staticanalysis/csharp/CSharpFileChecker.java

```diff
@@ -20,14 +20,17 @@
 import org.openrewrite.TreeVisitor;
 import org.openrewrite.csharp.tree.Cs;
 import org.openrewrite.marker.SearchResult;
+import org.openrewrite.staticanalysis.internal.ClassPathUtils;
 
 /**
  * Add a search marker if vising a CSharp file
  */
 public class CSharpFileChecker<P> extends TreeVisitor<Tree, P> {
+    private static final boolean IS_CSHARP_AVAILABLE = ClassPathUtils.isAvailable("org.openrewrite.csharp.tree.Cs$CompilationUnit");
```

Comment:
I'd perhaps keep it simple and just look for the outer wrapping class, not the inner class.
```suggestion
    private static final boolean IS_CSHARP_AVAILABLE = ClassPathUtils.isAvailable("org.openrewrite.csharp.tree.Cs");
```